tests/unit/OpenApiValidatorTest.php

```diff
@@ -88,6 +88,40 @@ public function testValidatorThrowsErrorWhenResponseIsInvalid(): void
         self::assertOpenApiSchema('tests/openapi.yaml', $browser);
     }
 
+    public function testValidatorThrowsErrorContainingInnerExceptionsWhenResponseFailsOneOfValidation(): void
+    {
+        $request = Request::create(uri: 'https://localhost/match-oneof', server: [
+            'HTTP_X_REQUESTED_WITH',
+            'XMLHttpRequest',
+        ]);
+        $response = new JsonResponse(['foo' => 'bar'], headers: ['content-type' => 'application/json']);
```

Comment:
Having a partial match is a better demonstration of your addition
```suggestion
        $response = new JsonResponse(['nested' => new \stdClass()], headers: ['content-type' => 'application/json']);
```